btree/btree.c

```diff
@@ -2908,16 +2942,13 @@ static int64_t btree_put_tick(struct m0_sm_op *smop)
 			struct slot    node_slot = {};
 			struct segaddr child_node_addr;
 
+			lev = &oi->i_level[oi->i_used];
 			lev->l_node = oi->i_nop.no_node;
 			node_slot.s_node = oi->i_nop.no_node;
 			lev->l_seq = lev->l_node->n_seq;
 			/* Verify node footer */
-			/*int rc = node_verify(lev->l_node);
-			if (rc)
-			{
-				node_op_fini(&oi->i_nop);
-				return fail(bop, rc);
-			}*/
+			M0_ASSERT(node_verify(lev->l_node));
```

Comment:
You should not assert because we do not want to crash the program. This check might fail at times and we just need to handle it. Like you can start over from the root to browse again.

Reply:
node_verify() will call m0_format_footer_verify(). m0_format_footer_verify() failure means that there is record corruption on the disk. Don't we want to crash in this case? restarting the state machine will result in the same failure. In the old btree code too m0_format_footer_verify() condition was part of M0_PRE().

Reply:
done